cmd/endpoint.go

```diff
@@ -200,6 +217,73 @@ func (endpoints EndpointList) GetString(i int) string {
 	return endpoints[i].String()
 }
 
+// UpdateIsLocal - resolves the host and discovers the local host.
+func (endpoints EndpointList) UpdateIsLocal() error {
+	var epsResolved int
+	var foundLocal bool
+	resolvedList := make([]bool, len(endpoints))
+	// Mark the starting time
+	startTime := time.Now()
+	keepAliveTicker := time.NewTicker(retryInterval * time.Second)
+	for {
+		// Break if the local endpoint is found already. Or all the endpoints are resolved.
+		if foundLocal || (epsResolved == len(endpoints)) {
+			break
+		}
+		// Retry infinitely on Kubernetes and Docker swarm.
+		// This is needed as the remote hosts are sometime
+		// not available immediately.
+		select {
+		case <-GlobalServiceDoneCh:
```

Comment:
are you sure that GlobalServiceDoneCh is already configured? if not you should look at 

```
globalOSSignalCh
```

as well and we should return an error here such that we can say "we were interrupted" - returning nil should be only used for success..

Reply:
No, the `GlobalServiceDoneCh` was not configured before. Neither `globalOSSignalCh`. I have to bring `signal.Notify(globalOSSignalCh, os.Interrupt, syscall.SIGTERM)` up early to use the `globalOSSignalCh`